iceoryx_posh/test/moduletests/test_mepoo_shm_safe_unmanaged_chunk.cpp

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/internal/mepoo/shm_safe_unmanaged_chunk.hpp"
+
+#include "iceoryx_posh/internal/mepoo/memory_manager.hpp"
+#include "iceoryx_posh/internal/mepoo/shared_chunk.hpp"
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+#include "iceoryx_posh/mepoo/mepoo_config.hpp"
+#include "iceoryx_utils/internal/posix_wrapper/shared_memory_object/allocator.hpp"
+
+#include "test.hpp"
+
+namespace
+{
+using namespace ::testing;
+
+using namespace iox::mepoo;
+
+class ShmSafeUnmanagedChunk_test : public Test
+{
+  public:
+    void SetUp() override
+    {
+        MePooConfig mempoolconf;
+        mempoolconf.addMemPool({CHUNK_SIZE, NUM_CHUNKS_IN_POOL});
+        memoryManager.configureMemoryManager(mempoolconf, m_memoryAllocator, m_memoryAllocator);
+    }
+
+    SharedChunk getChunkFromMemoryManager()
+    {
+        auto chunkSettingsResult = iox::mepoo::ChunkSettings::create(sizeof(bool), alignof(bool));
+        EXPECT_FALSE(chunkSettingsResult.has_error());
+        if (chunkSettingsResult.has_error())
+        {
+            return nullptr;
+        }
+        auto& chunkSettings = chunkSettingsResult.value();
+
+        return memoryManager.getChunk(chunkSettings);
+    }
+
+    iox::mepoo::MemoryManager memoryManager;
+
+  private:
+    static constexpr size_t KILOBYTE = 1 << 10;
+    static constexpr size_t MEMORY_SIZE = 100 * KILOBYTE;
+    std::unique_ptr<char[]> m_memory{new char[MEMORY_SIZE]};
+    static constexpr uint32_t NUM_CHUNKS_IN_POOL = 100;
+    static constexpr uint32_t CHUNK_SIZE = 128;
+
+    iox::posix::Allocator m_memoryAllocator{m_memory.get(), MEMORY_SIZE};
+};
+
+TEST_F(ShmSafeUnmanagedChunk_test, DefaultConstructedResultsInLogicallyNullptr)
+{
+    ShmSafeUnmanagedChunk sut;
+
+    EXPECT_TRUE(sut.isLogicalNullptr());
```

Comment:
You once suggest (also with an article) that we should test all side effects in one test. In this case I would add here that `getChunkHeader` returns the `nullptr` and all the other methods. This would reduce the tests a bit but I am still unsure if we should pursue this here.

Your approach is nice, readable and we can check that everything is covered but I wanted to mention this to you since it was a suggestion from you.